SQL/New_patches/2022-11-24-electrode-coord-system.sql

```diff
@@ -0,0 +1,252 @@
+  PRIMARY KEY (
+    `PhysiologicalCoordSystemID`,
+    `PhysiologicalElectrodeID`
+  ),
+  CONSTRAINT `FK_phys_coord_system_electrode_rel_coord_system`
+    FOREIGN KEY (`PhysiologicalCoordSystemID`)
+    REFERENCES `physiological_coord_system` (`PhysiologicalCoordSystemID`),
+  CONSTRAINT `FK_phys_coord_system_electrode_rel_electrode`
+    FOREIGN KEY (`PhysiologicalElectrodeID`)
+    REFERENCES `physiological_electrode` (`PhysiologicalElectrodeID`),
+  CONSTRAINT `FK_phys_coord_system_electrode_rel_phys_file`
+    FOREIGN KEY (`PhysiologicalFileID`)
+    REFERENCES `physiological_file` (`PhysiologicalFileID`)
+    ON DELETE CASCADE
+) ENGINE = InnoDB DEFAULT CHARSET = utf8;
+
+-- need insert before altering `physiological_electrode`
+-- migrate/init every coord system already present to 'not registered' state
+INSERT INTO physiological_coord_system_electrode_rel
+  (PhysiologicalCoordSystemID, PhysiologicalElectrodeID, PhysiologicalFileID, InsertTime)
+  SELECT DISTINCT
+    -- PhysiologicalCoordSystemID = not registered
+    (
+      SELECT pcs.PhysiologicalCoordSystemID
+      FROM physiological_coord_system AS pcs
+      INNER JOIN physiological_coord_system_name AS pcsn
+      WHERE pcsn.Name = 'Not registered'
+    ) AS 'PhysiologicalCoordSystemID',
+    -- PhysiologicalElectrodeID, PhysiologicalFileID, InsertTime
+    PhysiologicalElectrodeID,
+    PhysiologicalFileID,
+    InsertTime
+  FROM physiological_electrode;
+
+-- -----------------------------------------------------
+-- ALTERED
+-- Table `physiological_electrode`
+-- -----------------------------------------------------
+
+-- Table point3D manages coordinates now
+-- move coordinates to point_3d table
+
+-- copy distinct coordinate to `point_3d` table
+INSERT INTO point_3d
+  (X, Y, Z)
+  SELECT distinct X, Y, Z
+  FROM physiological_electrode;
+
+-- add the point3DID column to `physiological_electrode` table
+ALTER TABLE physiological_electrode
+  ADD Point3DID INT(10) UNSIGNED NOT NULL;
+
+-- update the point3DID in `physiological_electrode` table
+UPDATE point_3d as p, physiological_electrode as e
+  SET e.Point3DID = p.Point3DID
+  WHERE p.X = e.X
+    AND p.Y = e.Y
+    AND p.Z = e.Z;
+
+-- add foreign key to validate change
+ALTER TABLE physiological_electrode
+  ADD CONSTRAINT `FK_phys_electrode_point_3d`
+  FOREIGN KEY (`Point3DID`) REFERENCES `point_3d`(`Point3DID`);
+
+-- drop coordinate from `physiological_electrode` table
+ALTER TABLE physiological_electrode
+  DROP COLUMN X;
+ALTER TABLE physiological_electrode
+  DROP COLUMN Y;
+ALTER TABLE physiological_electrode
+  DROP COLUMN Z;
+
+
+-- `InsertTime` and `PhysiologicalFileID` will be moved
+-- to the relation table `physiological_coord_system_electrode_rel`
+ALTER TABLE physiological_electrode
+  DROP COLUMN InsertTime;
+
+ALTER TABLE physiological_electrode
+  DROP FOREIGN KEY FK_phys_file_FileID_3;
+ALTER TABLE physiological_electrode
+  DROP COLUMN PhysiologicalFileID;
```

Comment:
This line breaks functionality [here](https://github.com/aces/Loris/blob/main/modules/electrophysiology_browser/php/sessions.class.inc#L607)